model/task_lifecycle.go

```diff
@@ -776,6 +776,15 @@ func MarkEnd(ctx context.Context, settings *evergreen.Settings, t *task.Task, ca
 
 	detailsCopy := *detail
 	if t.ResultsFailed && detailsCopy.Status != evergreen.TaskFailed {
+		// TODO (DEVPROD-12889): remove this log and this logic once it's
+		// confirmed that all agents are on the newer version and the test
+		// results check has been moved.
+		grip.Debug(message.Fields{
+			"message": "overwriting task status to failed in app server due to test results containing failure",
+			"ticket":  "DEVPROD-12889",
+			"task_id": t.Id,
+			"host_id": t.HostId,
```

Comment:
Should we log what we got from the agent and what we are overwriting it to here as some bool (i.e. if the agent is getting the same result?)

Reply:
We could, but I don't think it would be that useful. The only statuses possible are success and failed, and in the case of success, the failing command type and description aren't meaningful anyways. I really just want this temporary log to confirm that the success isn't switched over to failure here (because it should already be done in the agent).